### ISSUE TITLE
Attempting to map a spanner array to a java array fails silently

### ISSUE THREAD

**Issue by Seth-Deal** [GoogleCloudPlatform/google-cloud-spanner-hibernate#231]
When using EntityManager to try and run a native query, the query fails silently when trying to map array columns. 

Example:
```
em.createNativeQuery(
            "SELECT " +
                "tile.tile_id as id, " +
                "tile.slug, " +
                "details.display_name, " +
                "tile.seller_user_name, " +
                "tile.listing_source, " +
                "tile.size, " +
                "tile.color, " +
                "tile.brand, " +
                "tile.condition  " +
                "FROM " +
                "tile_detail as details " +
                "join " +
                "tile AS tile ON tile.tile_id = details.tile_id " +
                filter +
                " LIMIT :limit", CatalogListing::class.java
        )
            .setParameter("limit", totalItems)
            .resultList
```
 Fails when trying to get the resultList. 
 
 My data model (CatalogListing) That i am trying to map back to is this:
 
 ```
 @Entity
data class CatalogListing(
    @Id
    val id: String,
    val slug: String,
    val displayName: String,
    val sellerUserName: String,
    val listingSource: String,
    val size: String,
    val color: Array<String>,
    val brand: Array<String>,
    val condition: String
) 
```

If i remove the two array columns, the call works again. 

```
cloudSpannerHibernateDialectVersion=1.4.0
cloudSpannerJdbcVersion=1.17.0
```

**Comment by dzou** [comment]
Thanks for the report. It looks like Hibernate [might not directly support ARRAY to ArrayList<T> mappings out-of-the-box](https://github.com/vladmihalcea/hibernate-types), but we might need to add some kind of adapter class to do this.

Will take a look and see if we can come up with a solution.

**Comment by dzou** [comment]
Collected some notes so far:
* tutorial on custom types: https://www.baeldung.com/hibernate-custom-types
* someone implemented Array type mappings for Postgres which is similar: https://github.com/vladmihalcea/hibernate-types

I think we need to do something like this; I'll double check with the Spanner client lib team and give it a try.

**Comment by dzou** [comment]
Feature is under way here: https://github.com/GoogleCloudPlatform/google-cloud-spanner-hibernate/pull/232

I got a proof of concept working; it's a little bit tricky though. Gonna add docs and samples and should be ready soon.

**Comment by dzou** [comment]
Just merged this in: https://github.com/GoogleCloudPlatform/google-cloud-spanner-hibernate/pull/232

I'll be doing a release shortly for this feature. If you want to try it immediately, you can depend on the `1.5.0-SNAPSHOT` version of the project. Add this repo to access the snapshot version:

```
<repository>
  <id>snapshots-repo</id>
  <url>https://oss.sonatype.org/content/repositories/snapshots</url>
  <releases><enabled>false</enabled></releases>
  <snapshots><enabled>true</enabled></snapshots>
</repository>
```

**Comment by dzou** [comment]
Thanks for your patience on this one -- we just released `1.5.0` so you'll be able to use this feature now. 

You can check out our [docs](https://github.com/GoogleCloudPlatform/google-cloud-spanner-hibernate#array-column-types) for an example and please feel free to post a question if you get stuck.

Also, one more note -- Regarding the silent failures, I believe Hibernate requires a `log4j.properties` file to have logging work properly; adding such a file to your `resources/` should hopefully reveal the log errors.

**Comment by Seth-Deal** [comment]
@dzou Thank you!
